trestle/core/utils.py

```diff
@@ -44,6 +44,88 @@ def get_elements_of_model_type(object_of_interest, type_of_interest):
     return loi
 
 
+def find_values_by_name_generic(object_of_interest, var_name):
+    """Traverse object and return list of the values in dicts, tuples associated with variable name."""
+    loe = []
+    # looking for a dict or 2-element tuple containing specified variable name
+    if type(object_of_interest) == dict:
+        for key, value in object_of_interest.items():
+            if (key == var_name) and value:
+                # found one so append its value to list
+                loe.append(value)
+            else:
+                new_list = find_values_by_name_generic(value, var_name)
+                if new_list:
+                    loe.extend(new_list)
+    elif type(object_of_interest) == tuple and len(object_of_interest) == 2 and object_of_interest[0] == var_name:
+        if object_of_interest[1]:
+            loe.append(object_of_interest[1])
+    elif type(object_of_interest) != str:
+        try:
+            # iterate over any iterable and recurse on its items
+            o_iter = iter(object_of_interest)
+        except Exception:
+            # it is not a dict and not iterable
+            pass
+        else:
+            next_item = next(o_iter, None)
+            while next_item is not None:
+                new_list = find_values_by_name_generic(next_item, var_name)
+                if new_list:
+                    loe.extend(new_list)
+                next_item = next(o_iter, None)
+            return loe
+    return loe
+
```

Comment:
So this one I'm a little concerned that this is going to do some slightly crazy things depending on type. I understand that there is an issue with the current 'finding' of objects of interest function which I wrote. This I'm worried that we are implicitly iterating over pydantic internals and I'm not sure what could happen.

would a 1/2 way approach be to twist the original search function to work by field name?

Reply:
I have one method that traverses object based on pydantic type - and it is similar to yours but in both of my methods I explicitly use the object's iterator to go through it.

But for the specific task of finding duplicate uuid's, the uuid's are simply string members of the class and I need to find them by name - i.e. use the generic search rather than pydantic.  I guess an alternative would be to scan a file like target.py and find all types that contain a uuid.  There are some actual uuid types like LocationUuid, but other uuid's are simply members of a class, like in Prop.  I would need to build a list of such objects - and search for all things like LocationUuid and Prop.  I now realize I am not catching LocationUuid's so I need to do that  - but accessing Prop.uuid would require building the list of classes that contain uuid.

Reply:
Lets discuss in the standup.